src/data/mv_polynomial/equiv.lean

```diff
@@ -319,6 +321,265 @@ begin
+  { have x : ¬ finsupp.cons i m = 0,
+    { apply finsupp.cons_nonzero_any,
+      simpa using c_i },
+    have x' :  ¬ 0 = finsupp.cons i m := by cc,
+    simp only [fin_succ_equiv_apply, coeff_C, function.comp_app, ring_hom.coe_comp, eval₂_hom_C,
+               polynomial.coeff_C, c_i, if_false, coeff_zero, x', if_false] },
+end
+
+private lemma fin_succ_equiv_coeff_coeff_case_p_X  {n : ℕ } {R : Type u} [comm_semiring R]
+  (p : mv_polynomial (fin (n + 1)) R) (j : fin (n + 1)) (hp : ∀ (i : ℕ) (m : fin n →₀ ℕ),
+  coeff m (((fin_succ_equiv R n) p).coeff i) = coeff (finsupp.cons i m) p) (i : ℕ)
+  (m : fin n →₀ ℕ) : coeff m (((fin_succ_equiv R n) (p * X j)).coeff i)
+  = coeff (finsupp.cons i m) (p * X j) :=
+begin
+  rw [coeff_mul_X' (finsupp.cons i m) j p, (fin_succ_equiv R n).map_mul],
+  by_cases c_j : j = 0,
+  { rw c_j,
+    by_cases c_i : i = 0,
+    { rw c_i,
+      simp only [fin_succ_equiv_apply, fin.cases_zero, polynomial.coeff_X_zero, coe_eval₂_hom,
+                  polynomial.mul_coeff_zero, finsupp.mem_support_iff, ne.def, eval₂_hom_X',
+                  mul_zero, coeff_zero ],
+      have t : ¬¬(finsupp.cons 0 m) 0 = 0 := by simp only [ not_not, finsupp.cons_zero],
+      simp only [t, if_false] },
+    { rw fin_succ_equiv_X_eq_zero,
+      let i' := nat.pred i,
+      have r : i = i'.succ,
+      { rw nat.succ_pred_eq_of_pos _,
+        exact nat.pos_of_ne_zero (by simpa) },
+      rw [r, polynomial.coeff_mul_X, hp i'],
+      simp only [finsupp.mem_support_iff, ne.def, finsupp.cons_zero],
+      rw ← r,
+      simp only [c_i, if_true, not_false_iff],
+      congr,
+      ext,
+      by_cases c_a : a = 0,
+      { rw c_a,
+        simp only [finsupp.single_eq_same, finsupp.coe_tsub, pi.sub_apply],
+        repeat { rw finsupp.cons_zero },
+        refl },
+      { simp only [finsupp.single_eq_same, finsupp.coe_tsub, pi.sub_apply],
+        have c_a' : a ≠ 0 := by simpa using c_a,
+        rw [←fin.succ_pred a c_a', finsupp.cons_succ, finsupp.cons_succ, finsupp.single],
+        have c_a'' : ¬ 0 = a := by simpa using c_a'.symm,
+        simp [c_a'', if_false] } } },
+  { let j' := fin.pred j c_j,
+    have r : j = j'.succ := by simp,
+    rw [r, fin_succ_equiv_X_ne_zero, polynomial.coeff_mul_C, coeff_mul_X', hp i],
+    simp only [fin.succ_pred, finsupp.mem_support_iff, ne.def],
+    by_cases c_mj' : m j' = 0,
+    { simp only [r, finsupp.cons_succ, c_mj', if_false, eq_self_iff_true, not_true] },
+    { simp only [r, finsupp.cons_succ, c_mj', if_true, not_false_iff],
+      congr,
+      ext,
+      by_cases c_a : a = 0,
+      { simp [c_a, finsupp.cons_zero, finsupp.single, c_j] },
+      { simp only [finsupp.coe_tsub, pi.sub_apply],
+        rw ←fin.succ_pred a c_a,
+        repeat {rw finsupp.cons_succ},
+        simp [finsupp.single, c_j],
+        congr } } },
+end
+
+/-- This gives the relation between the coefficients of the coefficients of `fin_succ_equiv R n f`
+    and the coefficients of `f`. -/
```

Comment:
```suggestion
/-- The relation between the coefficients of the coefficients of `fin_succ_equiv R n f`
    and the coefficients of `f`. -/
```
I think this comment can be more powerful if it actually says what it does. Right now the user will need to inspect the statement anyway to know what the relation is. I don't think this comment adds anything right now that can not be gained from a quick look at the type.